pylint/checkers/classes/class_checker.py

```diff
@@ -1375,7 +1375,8 @@ def _check_redefined_slots(
     ) -> None:
         """Check if `node` redefines a slot which is defined in an ancestor class"""
         slots_names = [
-            slot.value for slot in slots_list if isinstance(slot, nodes.Const)
+            slot.value if isinstance(slot, nodes.Const) else next(slot.infer()).value
```

Comment:
```suggestion
            slot.value if isinstance(slot, nodes.Const) else utils.safe_infer(slot).value
```

Reply:
Using `getattr` since have the following from mypy otherwise:
`pylint/checkers/classes/class_checker.py:1378: error: Item "None" of "Optional[Any]" has no attribute "value"  [union-attr]`